### ISSUE TITLE
missing libpthread

### ISSUE THREAD

**Issue by drewp** [vsergeev/audioprism#1]
On ubuntu 16.04:

make fails:

```
% make
g++ build/src/audio/PulseAudioSource.o build/src/audio/WaveAudioSource.o build/src/dft/RealDft.o build/src/image/MagickImageSink.o build/src/spectrogram/SpectrumRenderer.o build/src/main/AudioThread.o build/src/main/SpectrogramThread.o build/src/main/InterfaceThread.o build/src/main/main.o -o audioprism -lpulse-simple -lpulse -lfftw3 -lsndfile -lSDL2_ttf -lSDL2 -lGraphicsMagick++ -lGraphicsMagick
/usr/bin/ld: build/src/main/AudioThread.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:63: recipe for target 'audioprism' failed
make: *** [audioprism] Error 1
```

But it works with -lpthread:

```
% g++ build/src/audio/PulseAudioSource.o build/src/audio/WaveAudioSource.o build/src/dft/RealDft.o build/src/image/MagickImageSink.o build/src/spectrogram/SpectrumRenderer.o build/src/main/AudioThread.o build/src/main/SpectrogramThread.o build/src/main/InterfaceThread.o build/src/main/main.o -o audioprism -lpulse-simple -lpulse -lfftw3 -lsndfile -lSDL2_ttf -lSDL2 -lGraphicsMagick++ -lGraphicsMagick -lpthread   
%
```

**Comment by AlexGustafsson** [comment]
Solution:

Edit the Makefile to reflect the following changes.

(Makefile lines 24-34)

```
################################################################################

CXX = g++
REMOVE = rm -rf

CPPFLAGS = -std=c++11 -W -Wall -Wextra -Wconversion -pedantic -O3 -g -Isrc/
CPPFLAGS += $(shell pkg-config --cflags libpulse libpulse-simple fftw3 sndfile sdl2 SDL2_ttf GraphicsMagick++)

LDFLAGS = $(shell pkg-config --libs libpulse libpulse-simple fftw3 sndfile sdl2 SDL2_ttf GraphicsMagick++)

################################################################################
```

to

(Makefile lines 24-34)

```
################################################################################

CXX = g++
REMOVE = rm -rf

CPPFLAGS = -std=c++11 -W -Wall -Wextra -Wconversion -pedantic -O3 -g -Isrc/
CPPFLAGS += $(shell pkg-config --cflags libpulse libpulse-simple fftw3 sndfile sdl2 SDL2_ttf GraphicsMagick++)

LDFLAGS = $(shell pkg-config --libs libpulse libpulse-simple fftw3 sndfile sdl2 SDL2_ttf GraphicsMagick++) -lpthread

################################################################################
```

The change is the addition of `-lpthread` on the end of line 32 (LDFLAGS)

**Comment by vsergeev** [comment]
Thanks guys. I've fixed this in fc6507f and released a patch release with the fix.
